### PR TITLE
Handle unix signals

### PR DESCRIPTION
**Pull request by cloudhead**

### CODE REVIEW

File: radicle-node/src/signals.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    }
+    Ok(())
+}
+
+/// Install global signal handlers for `SIGTERM` and `SIGINT`.
+///
+/// # Safety
+///
+/// Calls `libc` functions safely.
+unsafe fn _install() -> io::Result<()> {
```
Comment (slack-coder) [inline radicle-node/src/signals.rs:35]: why the underscore?

File: radicle-node/src/signals.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        }
+        *channel = Some(notify);
+
+        unsafe { _install() }?;
+    } else {
+        return Err(io::Error::new(
+            io::ErrorKind::WouldBlock,
+            "unable to install signal handler",
+        ));
+    }
```
Comment (slack-coder) [inline radicle-node/src/signals.rs:26]: What about lifting the nested if statement using the new let statements?

i.e.
```
let Ok(mut channel) = ... else {
  return Err(..)
} 
```

**Comment by slack-coder** [comment]
It could be possible to have the signals clearly mapped to their actions, maybe in main?  However it may take some refactoring, therefore as long as this works, its good for me.

File: radicle-node/src/signals.rs (hunk 1 above)
Comment (cloudhead) [reply radicle-node/src/signals.rs:35]: There's another function called `install` that calls this function :) It's a convention I've seen in a bunch of places when you have an inner and an outer function.

File: radicle-node/src/signals.rs (hunk 2 above)
Comment (cloudhead) [reply radicle-node/src/signals.rs:26]: Tbh I don't think it really makes a difference here, the function is small..

**Comment by cloudhead** [comment]
Yeah, I opted for the signal setup to be in main, while the signal handling logic is in the library code. This allows triggering the signal handler in tests by sending a message on the channel.

**Review by adaszko — APPROVED** [review]
Looks good 👍

**Review by FintanH — APPROVED** [review]
LGTM

**Pull request merged**